lib/dialects/mariadb/query.js

```diff
@@ -183,11 +183,12 @@ class Query extends AbstractQuery {
       if (modelField.type instanceof DataTypes.JSON) {
         // Value is returned as String, not JSON
         rows = rows.map(row => {
-          row[modelField.fieldName] = row[modelField.fieldName] ? JSON.parse(
-            row[modelField.fieldName]) : null;
-          if (DataTypes.JSON.parse) {
-            return DataTypes.JSON.parse(modelField, this.sequelize.options,
-              row[modelField.fieldName]);
+          if(row[modelField.fieldName] !== undefined) {
```

Comment:
oh also: it looks like this doesn't fix #13027.

for that you should probably do something like:
```suggestion
          if(typeof row[modelField.fieldName] === 'string') {
```